service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -43,4 +43,23 @@
      * A list of service URIs.
      */
     public abstract List<String> uris();
+
+    // hides implementation details
+    public static Builder builder() {
+        return ImmutableServiceConfiguration.builder();
+    }
+
+    // hides implementation details
+    public interface Builder {
```

Comment:
Any volunteer for making this change here?

On Tue, Apr 26, 2016 at 8:21 AM Nick Miyake <notifications@github.com>
wrote:

> In
> service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java
> <https://github.com/palantir/http-remoting/pull/106#discussion_r61038347>:
>
> > @@ -43,4 +43,23 @@
> >       * A list of service URIs.
> >       */
> >      public abstract List<String> uris();
> > +
> > +    // hides implementation details
> > +    public static Builder builder() {
> > +        return ImmutableServiceConfiguration.builder();
> > +    }
> > +
> > +    // hides implementation details
> > +    public interface Builder {
>
> Sorry to keep harping on this thread, but adding some notes based on what
> I just figured out myself.
>
> The approach that Rob and Paul suggested never occurred to me because
> originally the generated Builder class is final, so never thought that
> subclassing it was a possibility. It seems that you need to include the Builder
> builder() method and public static final class Builder extends
> Immutable*.Builder {} code in the original in order for the generated
> code to work properly. However, once that's done, everything does seem to
> work for me.
>
> This definitely seems more elegant from a code standpoint, so I'm probably
> going to move to this approach for my own code. Thanks for the tip!
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/palantir/http-remoting/pull/106/files/511164691ff08f26a4da8c6c8ee8e99f7521bccf#r61038347>
>